auth/user_mgt_test.go

```diff
@@ -0,0 +1,704 @@
+		if err != nil {
+			t.Fatal(err)
+		}
+	default:
+		b, err = json.Marshal(resp)
+		if err != nil {
+			t.Fatal("marshaling error")
+		}
+	}
+
+	s := mockAuthServer{Resp: b}
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		s.Req = r
+		for k, v := range s.Header {
+			w.Header().Set(k, v)
+		}
+		if s.Status != 0 {
+			w.WriteHeader(s.Status)
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(s.Resp)
+
+	})
+	s.srv = httptest.NewServer(handler)
+	authClient, err := NewClient(context.Background(),
+		&internal.AuthConfig{
+			Opts: []option.ClientOption{option.WithHTTPClient(s.srv.Client())},
+		})
+	if err != nil {
+		t.Error()
+	}
+	authClient.url = s.srv.URL + "/"
+	s.client = authClient
+	return &s, s.srv.Close
+}
+
+func (s *mockAuthServer) Client() *Client {
+	return s.client
+}
+
+/*func TestSetClaimsField(t *testing.T) {
+	tests := []struct {
+		p          UserParams
+		attributes string
+	}{
+		{
+			UserParams{
+				CustomClaims: map[string]interface{}{
+					"asdf":  "ff",
+					"asdff": true,
+				},
+			},
+			"{\"asdf\":\"ff\",\"asdff\":true}",
+		}, {
+			UserParams{
+				CustomClaims: map[string]interface{}{},
+			},
+			"{}",
+		}, {
+			UserParams{
+				CustomClaims: map[string]interface{}{
+					"integer": 12,
+					"number":  12.3,
+				},
+			},
+			"{\"integer\":12,\"number\":12.3}",
+		},
+	}
+	for _, test := range tests {
+		up := userParams{UserParams: &test.p}
+
+		up.setClaimsField()
+		if *up.CustomAttributes != test.attributes {
+			t.Errorf("wrong custom attribute string got %s expecting %s", *up.CustomAttributes, test.attributes)
+
+		}
+	}
+}
+
+func TestDeleteParams(t *testing.T) {
+	tests := []struct {
+		params UserParams
+		up     userParams
+	}{
+		{
+			UserParams{
+				DisplayName: ptr.String(""),
+			},
+			userParams{
+				UserParams:          &UserParams{},
+				DeleteAttributeList: []string{"DISPLAY_NAME"},
+			},
+		},
+		{
+			UserParams{
+				PhotoURL: ptr.String(""),
+			},
+			userParams{
+				UserParams:          &UserParams{},
+				DeleteAttributeList: []string{"PHOTO_URL"},
+			},
+		}, {
+			UserParams{
+				DisplayName: ptr.String(""),
+				PhotoURL:    ptr.String(""),
+			},
+			userParams{
+				UserParams:          &UserParams{},
+				DeleteAttributeList: []string{"DISPLAY_NAME", "PHOTO_URL"},
+			},
+		},
+		{
+			UserParams{
+				PhoneNumber: ptr.String(""),
+			},
+			userParams{
+				UserParams:         &UserParams{},
+				DeleteProviderList: []string{"phone"},
+			},
+		},
+	}
+	for i, test := range tests {
+		up := userParams{UserParams: &test.params}
+		up.setDeleteFields()
+		if !reflect.DeepEqual(up, test.up) {
+			t.Errorf("test %d: got %#v expecting %#v", i, up, test.up)
+		}
+	}
+}
+*/
+func TestGetUser(t *testing.T) {
+	s, closer := echoServer("get_user.json", t)
+	defer closer()
+
+	user, err := s.Client().GetUser(context.Background(), "ignored_id")
+	if err != nil {
+		t.Error(err)
+	}
+	tests := []struct {
```

Comment:
This way of testing doesn't make much sense for the scenario at hand. I recommend you break this up into a series of if statements. You can also possibly wrap them in helper function, if you do this type of checks elsewhere.